### PR TITLE
Create room invite delete

### PR DESCRIPTION
**Pull request by ghost**
Removed "sleeps", made independence feature

### CODE REVIEW

**Review by helltepakcer — APPROVED** [review]
You should to delete your password from files, CI can work without that, I just upload file like in master. P.S. I done that after tip from Sergey

File: features/steps/step_selenium_hipchat_create_room.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    context.lobby_page.click_add_member()
+    context.wait.until(EC.element_to_be_clickable((By.XPATH, '//*[@id="page"]/div[1]/div[2]/div/div[2]/div[2]/div/div[1]/div[2]/div[2]/div[3]/div/div/div[2]/div[2]/a')))
+    context.lobby_page.click_ask_to_join()
+    context.wait.until(EC.element_to_be_clickable((By.CSS_SELECTOR, '#s2id_invite-users-people')))
+    context.lobby_page.send_invite()
+    context.wait.until(EC.element_to_be_clickable((By.XPATH, '//*[@id="invite-users-dialog"]/footer/div[1]/button[1]')))
+    context.lobby_page.invite()
+
+
+@then('we relogin')
```
Comment (SergeyMoroz0703) [inline features/steps/step_selenium_hipchat_create_room.py]: in your `feature` you put `@and` here instead `@then` and check in another lines the same issue

File: .travis.yml [hunk 2]
```diff
@@ -13,4 +17,4 @@ install:
   - pip install -r requirements.txt
 script:
   - python --version
-  - behave
+  - behave
```
Comment (timbortnik) [inline .travis.yml]: Is there a reason to remove newline at end of file?

File: features/environment.py [hunk 3]
```diff
@@ -30,10 +32,13 @@ def before_all(context):
 
     context.base_url = "https://www.hipchat.com"
     context.driver = webdriver.Chrome()
+    context.wait = ui.WebDriverWait(context.driver, 10)
 
     context.login_page = LoginPage(context)
     context.authorized_page = AuthorizedPage(context)
+    context.lobby_page = LobbyPage(context)
+
```
Comment (timbortnik) [inline features/environment.py]: Could you please "inspect code" for PEP8 warnings? Like extra lines, etc.

File: features/environment_secret.py [hunk 4]
```diff
@@ -1,5 +1,7 @@
 # -*- coding: UTF-8 -*-
 import os
 
-HIPCHAT_LOGIN = os.environ.get("HCLOGIN", "tim+test@bortnik.org")
-HIPCHAT_PASS = os.environ.get("HCPASS", "ntcn987654321")
+HIPCHAT_LOGIN = os.environ.get("HCLOGIN", "ighost1996@gmail.com")
+HIPCHAT_PASS = os.environ.get("HCPASS", "qwert12345")
```
Comment (timbortnik) [inline features/environment_secret.py]: Please use environment variables, any password that was pushed to open repo should be changed ASAP

File: features/steps/step_selenium_hipchat.py [hunk 5]
```diff
@@ -7,6 +7,9 @@
 from behave import given, when, then
 
 
+
```
Comment (timbortnik) [inline features/steps/step_selenium_hipchat.py]: Do we need extra lines here?

File: features/steps/step_selenium_hipchat_create_room.py [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+# ----------------------------------------------------------------------------
+from behave import given, when, then
+from selenium.webdriver.support import expected_conditions as EC
+from selenium.webdriver.common.by import By
+
+
+@given('we are on Hipchat Lobby Page')
+def step_impl(context):
+    context.authorized_page.enter_app()
+    pass
```
Comment (timbortnik) [inline features/steps/step_selenium_hipchat_create_room.py]: "pass" looks like redundant here

File: features/steps/step_selenium_hipchat_create_room.py [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+    context.lobby_page.click_ask_to_join()
+    context.wait.until(EC.element_to_be_clickable((By.CSS_SELECTOR, '#s2id_invite-users-people')))
+    context.lobby_page.send_invite()
+    context.wait.until(EC.element_to_be_clickable((By.XPATH, '//*[@id="invite-users-dialog"]/footer/div[1]/button[1]')))
+    context.lobby_page.invite()
+
+
+@then('we relogin')
+def step_impl(context):
+    context.lobby_page.renavigate()
```
Comment (timbortnik) [inline features/steps/step_selenium_hipchat_create_room.py]: what is the difference between this code and "context.authorized_page.enter_app()"?

Comment (timbortnik) [reply features/steps/step_selenium_hipchat_create_room.py]: https://en.wikipedia.org/wiki/Don%27t_repeat_yourself - this code looks redundant

File: features/steps/step_selenium_hipchat_create_room.py [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+    context.login_page.login()
+    assert "Welcome," in context.authorized_page.get_page_head()
+    context.authorized_page.enter_app()
+
+@then('we accept the invitation')
+def step_impl(context):
+    context.lobby_page.accept_invite()
+
+@then('we relogin again')
+def step_impl(context):
```
Comment (timbortnik) [inline features/steps/step_selenium_hipchat_create_room.py]: This code looks redundant

File: .travis.yml [hunk 9]
```diff
@@ -1,4 +1,8 @@
 dist: trusty
+sudo: required
+
+addons:
+  chrome: stable
```
Comment (timbortnik) [inline .travis.yml]: I would suggest to merge from master. Copying changes manually may lead to merge conflicts.

**Review by timbortnik — CHANGES_REQUESTED** [review]

File: features/steps/step_selenium_hipchat_create_room.py (hunk 7 above)
Comment (ghost) [reply features/steps/step_selenium_hipchat_create_room.py]: Here we go to the /sign_in page and relogin to accept the invitation. In "context.authorized_page.enter_app()" we enter the app by clicking the "launch app button"

File: features/steps/step_selenium_hipchat_create_room.py (hunk 1 above)
Comment (ghost) [reply features/steps/step_selenium_hipchat_create_room.py]: answered in slack

File: features/steps/step_selenium_hipchat_create_room.py [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+    context.authorized_page.enter_app()
+    pass
+
+@then('we create a room')
+def step_impl(context):
+    context.lobby_page.create_room()
+    context.wait.until(EC.element_to_be_clickable((By.ID, 'create-room-name')))
+    context.lobby_page.set_name()
+    context.lobby_page.click_create_room()
+    context.wait.until(EC.element_to_be_clickable((By.XPATH, '//*[@id="page"]/div[1]/div[2]/div/div[2]/div[2]/div/div[1]/div[2]/div[2]/ul/li[1]/div')))
```
Comment (helltepakcer) [inline features/steps/step_selenium_hipchat_create_room.py]: Try to find other solution to find your element, this code to long, in our Slack - materials - I had downloaded file with different hooks to find element with more accurate code, with this hooks I removed my "long tails" search.

File: pages/lobby_page.py [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+
+
+    def accept_invite(self):
+        self.context.driver.find_element_by_xpath('//*[@id="page"]/div[1]/div[2]/div/div/div/div[1]').click()
+        from selenium.webdriver.common.keys import Keys
+        self.context.driver.find_element_by_id('hc-message-input').send_keys('@all', Keys.RETURN, Keys.RETURN)
+        self.context.wait.until(EC.visibility_of_element_located((By.XPATH, '//*[@class="msg-line"]')))
+
+
+
```
Comment (helltepakcer) [inline pages/lobby_page.py]: Please remove empty lines as specified in PEP8.

File: features/steps/step_selenium_hipchat_create_room.py (hunk 6 above)
Comment (ghost) [reply features/steps/step_selenium_hipchat_create_room.py]: Fixed

File: features/steps/step_selenium_hipchat_create_room.py (hunk 8 above)
Comment (ghost) [reply features/steps/step_selenium_hipchat_create_room.py]: Fixed

File: features/steps/step_selenium_hipchat_create_room.py (hunk 10 above)
Comment (ghost) [reply features/steps/step_selenium_hipchat_create_room.py]: Fixed smth

File: pages/lobby_page.py (hunk 11 above)
Comment (ghost) [reply pages/lobby_page.py]: Fixed

**Review by timbortnik — APPROVED** [review]

File: features/steps/step_selenium_hipchat.py [hunk 12]
```diff
@@ -27,7 +27,22 @@ def step_impl(context):
     pass
 
 
+@when('we enter other login')
+def step_impl(context):
+    context.login_page.enter_login(context.hipchat_login_2)
+    context.login_page.login()
+    pass
```
Comment (timbortnik) [inline features/steps/step_selenium_hipchat.py:34]: No need to put "pass" here

File: features/steps/step_selenium_hipchat.py [hunk 13]
```diff
@@ -27,7 +27,22 @@ def step_impl(context):
...
+    context.login_page.enter_pass(context.hipchat_pass_2)
+    context.login_page.login()
+    pass
+
+
 @then('we see welcome title')
 def step_impl(context):
     assert "Welcome," in context.authorized_page.get_page_head()
 
+
```
Comment (timbortnik) [inline features/steps/step_selenium_hipchat.py]: Please, do not add unnecessary lines to your PRs

File: features/steps/step_selenium_hipchat_create_room.py [hunk 14]
```diff
@@ -0,0 +1,36 @@
...
+# @mark.steps
+# ----------------------------------------------------------------------------
+# STEPS:
+# ----------------------------------------------------------------------------
+from behave import given, when, then
+
+
+@given('we are on Hipchat Lobby Page')
+def step_impl(context):
+    context.authorized_page.enter_app()
```
Comment (timbortnik) [inline features/steps/step_selenium_hipchat_create_room.py:12]: DRY? Not sure, but do we have similar step elsewhere?

File: pages/lobby_page.py [hunk 15]
```diff
@@ -0,0 +1,89 @@
...
+    def find_add_member(self):
+        return self.context.driver.find_element_by_xpath('//*[@class="hc-glance clickable"]')
+
+    def click_add_member(self):
+        import time
+        time.sleep(3)
+        self.context.driver.find_element_by_id('room-actions-btn').click()
+        self.context.wait.until(EC.element_to_be_clickable((By.XPATH, '//a[text()="Invite People"]')))
+        self.context.driver.find_element_by_xpath('//a[text()="Invite People"]').click()
+
```
Comment (timbortnik) [inline pages/lobby_page.py:51]: PEP8?

**Review by timbortnik — APPROVED** [review]

**Pull request merged**